pywb/static/wombat.js

```diff
@@ -2856,7 +2856,72 @@ var _WBWombat = function($wbwindow, wbinfo) {
         wb_funToString.apply = orig_apply;
     }
 
-
+    function initTimeoutIntervalOverrides($wbwindow) {
+        // because [setTimeout|setInterval]('document.location.href = "xyz.com"', time) is legal and used
+        // declare the beginning wrapper string first so we do not redeclare it multiple time in our override (closure)
+        var wrapper = 'var _____WB$wombat$assign$function_____ = function(name) {return (self._wb_wombat && self._wb_wombat.local_init &&self._wb_wombat.local_init(name)) || self[name]; };\n' +
+            'if (!self.__WB_pmw) { self.__WB_pmw = function(obj) { return obj; } }\n{\n' +
+            'let window = _____WB$wombat$assign$function_____("window");\n' +
+            'let self = _____WB$wombat$assign$function_____("self");\n' +
+            'let document = _____WB$wombat$assign$function_____("document");\n' +
+            'let location = _____WB$wombat$assign$function_____("location");\n' +
+            'let top = _____WB$wombat$assign$function_____("top");\n' +
+            'let parent = _____WB$wombat$assign$function_____("parent");\n' +
+            'let frames = _____WB$wombat$assign$function_____("frames");\n' +
+            'let opener = _____WB$wombat$assign$function_____("opener");\n';
+
+        if ($wbwindow.setTimeout && !$wbwindow.setTimeout.__$wbpatched$__) {
+            var originalSTO = $wbwindow.setTimeout;
+            $wbwindow.setTimeout = function () {
+                // strings are primitives with a prototype or __proto__ of String depending on the browser
+                var rw = arguments[0] != null && Object.getPrototypeOf(arguments[0]) === String.prototype;
+                // do not mess with the arguments object unless you want instant de-optimization
+                var args = rw ? new Array(arguments.length) : arguments;
+                if (rw) {
+                    if ($wbwindow.Proxy) {
+                        args[0] = wrapper + arguments[0] + '\n}';
+                    } else {
+                        args[0] = arguments[0].replace(/\blocation\b/g, "WB_wombat_$&");
+                    }
+                    for (var i = 1; i < arguments.length; ++i) {
+                        args[i] = proxy_to_obj(arguments[i]);
+                    }
+                }
+                // setTimeout does not require its this arg to be window so just in case someone got funky with it
+                if (originalSTO.__WB_orig_apply) {
+                    return originalSTO.__WB_orig_apply(null, args)
```

Comment:
Think we just need to add proxy_to_obj here as well and then it'll work. Seems safer than null:
`originalSTO.__WB_orig_apply(proxy_to_obj(this), args)`

Reply:
Ah wow, nice find/docs of the issue! I wonder if something else is the cause of this exact issue, but should be good for now. Something to keep in mind if we see this again.